### PR TITLE
Add route speed functions

### PR DESCRIPTION
**Pull request by elliotjo**

### CODE REVIEW

**Comment by malban** [comment]
Can you rebase this with master?

**Comment by elliotjo** [comment]
Rebased

File: swri_route_util/CMakeLists.txt [hunk 1]
```diff
@@ -4,6 +4,8 @@ project(swri_route_util)
 set(BUILD_DEPS 
   marti_nav_msgs
```
Comment (evenator) [inline swri_route_util/CMakeLists.txt]: It looks like you're missing dependencies on `marti_common_msgs`, `swri_route_util`, and `visualization_msgs`.

File: swri_route_util/package.xml [hunk 2]
```diff
@@ -16,6 +16,8 @@
   <depend>marti_nav_msgs</depend>
   <depend>roscpp</depend>
   <depend>swri_transform_util</depend>
+  <depend>swri_math_util</depend>
+  <depend>swri_geometry_util</depend>
```
Comment (evenator) [inline swri_route_util/package.xml]: See my comment on `CMakeLists.txt` about missing dependencies.

File: swri_route_util/include/swri_route_util/route_speeds.h [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  bool use_speed_from_accel_constant_;
+  /// Maximum lateral acceleration in accel mode in m/s^2
+  double max_lateral_accel_mss_;
+
+  /// Speed as a function of curvature, applies when
+  /// use_speed_from_accel_constant is false
+  swri_math_util::Interpolation1D speed_curve_;
+
+  // Filter constant used when estimating route curvature.  Larger
+  // values result in smoother curvature estiamtes with fewer spikes.
```
Comment (evenator) [inline swri_route_util/include/swri_route_util/route_speeds.h]: Typo here "estiamtes"

**Comment by elliotjo** [comment]
Addressed Ed's comments and rebased again.

**Comment by malban** [comment]
Is the build failure unrelated?

**Comment by evenator** [comment]
@malban It looks like we have some flaky tests in `swri_transform_util`. I'll rerun the failed builds.

**Comment by evenator** [comment]
I'm tracking the flaky test in #468, if @pjreed or @malban is looking for a challenge.

**Pull request merged**